crl_test.go

```
@@ -888,34 +888,36 @@ func closeServer(t *testing.T, server *http.Server) {
 func TestCrlE2E(t *testing.T) {
 	crlCacheCleaner.stopPeriodicCacheCleanup()
 	t.Run("Successful flow", func(t *testing.T) {
-		_ = logger.SetLogLevel("debug")
-		defer func() {
-			_ = logger.SetLogLevel("error")
-		}()
+		skipOnJenkins(t, "CRL E2E test requires real Snowflake certificates with CRL distribution points")
+
 		cleanupCrlCache(t)
 		previousCrlCacheCleanerTickRate := crlCacheCleanerTickRate
 		previousCacheValidityTime := crlCacheCleaner.cacheValidityTime
 		defer func() {
 			crlCacheCleanerTickRate = previousCrlCacheCleanerTickRate
 			crlCacheCleaner.cacheValidityTime = previousCacheValidityTime
 			crlCacheCleaner.stopPeriodicCacheCleanup()
+			createDSN("UTC") // Restore global DSN after test
 		}()
 		crlCacheCleanerTickRate = 5 * time.Second
 		crlCacheCleaner.cacheValidityTime = 60 * time.Second
-		cfg := &Config{
-			User:                              username,
-			Password:                          pass,
-			Account:                           account,
-			Database:                          dbname,
-			Schema:                            schemaname,
-			CertRevocationCheckMode:           CertRevocationCheckEnabled,
-			CrlAllowCertificatesWithoutCrlURL: ConfigBoolTrue,
-			DisableOCSPChecks:                 true,
+
+		// Use the global DSN with real JWT/private key configuration
+		createDSN("UTC")
+		crlDSN := dsn + "&certRevocationCheckMode=ENABLED&crlAllowCertificatesWithoutCrlURL=true&disableOCSPChecks=true"
+
+		db, err := sql.Open("snowflake", crlDSN)
+		if err != nil {
+			maskedErr := maskSecrets(err.Error())
+			t.Fatalf("CRL E2E test failed to open connection: %s", maskedErr)
 		}
-		db := sql.OpenDB(NewConnector(SnowflakeDriver{}, *cfg))
 		defer db.Close()
 		rows, err := db.Query("SELECT 1")
-		assertNilF(t, err)
+		if err != nil {
+			// Mask any sensitive information in error messages before failing
+			maskedErr := maskSecrets(err.Error())
+			t.Fatalf("CRL E2E test failed: %s", maskedErr)
```

I expect that you use custom error, because our assertions don't mask errors. Maybe we should apply masking there and here use them?

---

replaced with assert